### ISSUE TITLE
Yolov8 build a new model from scratch not able to improve accuracy ?

### ISSUE THREAD

**Issue by TapendraBaduwal** [ultralytics/ultralytics#2456]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I just try with your coco128 dataset like this but model just give small accuracy like 10 to 12 % ?

I just test below code not add any thing from outside.

model = YOLO('yolov8n.yaml')  # build a new model from scratch
results = model.train(data='coco128.yaml', epochs=200,lr0=0.001)  # train the model

**Comment by glenn-jocher** [comment]
@TapendraBaduwal hello there,

Thank you for trying out YOLOv8 and using the coco128 dataset. If you're only getting 10-12% accuracy, it's possible that there might be some issues with the data or training parameters. The coco128 dataset is relatively small, so it's important to ensure that the data is balanced and representative of the classes you're interested in detecting. 

For the training parameters, you might want to experiment with different learning rates, batch sizes, and number of epochs. Sometimes, adjusting these hyperparameters can significantly improve the model's accuracy. 

Lastly, it's also possible that the model architecture needs to be modified to better suit your needs. YOLOv8 has a lot of flexibility in terms of customizing the model architecture through the YAML file. 

I hope these suggestions help, and please let us know if you have any further questions or issues with YOLOv8.

Best regards,
Glenn Jocher

**Comment by TapendraBaduwal** [comment]
@glenn-jocher I set 500 epoch but it break out with 150 epoch ...how to solve this problem?

**Comment by glenn-jocher** [comment]
Hi @TapendraBaduwal-htike,

Thanks for reaching out. If your training process seems to be terminating prematurely, there could be a few possible causes. Here are some things you can try to help diagnose and solve the issue:

1. Check for any error messages or logs during the training process. This could give you some insights into what's causing the issue. If there are any error messages, please let me know and I can help troubleshoot further.

2. Check your training data: make sure that there are no missing or corrupt files that could be causing issues. Additionally, you might want to check that your data is well-balanced across classes.

3. Adjust the `batch-size` and reduce the number of epochs to see if the training process runs to completion. It's possible that your system may not have enough resources to train with a large batch size or some of the training data might be difficult to train on.

Hopefully these suggestions help. Once you've narrowed down the issue, please feel free to reach out again and I'll be happy to help some more.

Best regards,
Glenn Jocher

**Comment by Donghyun-Son** [comment]
> @glenn-jocher I set 500 epoch but it break out with 150 epoch ...how to solve this problem?

I think you need to check patience.

**Comment by glenn-jocher** [comment]
Hello @Donghyun-Son, 

Thank you for your question. Regarding the issue with breaking out of training at 150 epochs, it's possible that the training is ending early due to a low patience value. The patience value in the training process determines the number of epochs that the model can stall without any improvement in loss before it stops training. 

If patience is set too low, it can cause the model to stop training before it has fully converged to a good solution. Therefore, it's recommended to increase the patience value if you are experiencing the issue of training stopping before you reach the desired number of epochs. 

I hope this helps! Please let me know if you have any further questions or concerns. 

Best regards, 
Glenn Jocher
